Orange/data/table.py

```diff
@@ -523,13 +525,51 @@ def from_file(filename, wrapper=None):
 
     @classmethod
     def from_url(cls, url):
-        name = os.path.basename(urllib.parse.urlparse(url)[2])
-        f = tempfile.NamedTemporaryFile(suffix=name, delete=False)
-        fname = f.name
-        f.close()
-        urllib.request.urlretrieve(url, fname)
-        data = cls.from_file(f.name)
-        os.remove(fname)
+        # Resolve (potential) redirects to a final URL
+        response = urlopen(url, timeout=10)
+        url = response.url
+        response.close()
+
+        def url_googlesheets(url):
+            match = re.match(r'(?:https?://)?(?:www\.)?'
+                             'docs\.google\.com/spreadsheets/d/'
+                             '(?P<workbook_id>[-\w_]+)'
+                             '(?:/.*?gid=(?P<sheet_id>\d+).*|.*)?',
+                             url, re.IGNORECASE)
+            try:
+                workbook, sheet = match.group('workbook_id'), match.group('sheet_id')
+                if not workbook: raise ValueError
+            except (AttributeError, ValueError):
+                raise ValueError
+            url = 'https://docs.google.com/spreadsheets/d/{}/export?format=tsv'.format(workbook)
+            if sheet: url += '&gid=' + sheet
+            return url
+
+        URL_TRIMMERS = (
+            url_googlesheets,
+        )
+        for trim in URL_TRIMMERS:
```

Comment:
PEP 8:

While sometimes it's okay to put an if/for/while with a small body on the same line, never do this for multi-clause statements. Also avoid folding such long lines!

Rather not:

if foo == 'blah': do_blah_thing()

Definitely not:

if foo == 'blah': do_blah_thing()
else: do_non_blah_thing()

try: something()
finally: cleanup()